src/parsec.rs

```diff
@@ -296,14 +311,25 @@ impl ParsecMap {
 
 // Generate a DkgResult that can be used for the first node
 pub fn generate_first_dkg_result(rng: &mut MainRng) -> DkgResult {
-    let threshold = 0;
+    let participants = 1;
     let index = 0;
 
-    let secret_key_set = RealBlsSecretKeySet::random(threshold, &mut RngCompat(rng));
+    let secret_key_set = generate_bls_threshold_secret_key(rng, participants);
     let secret_key_share = secret_key_set.secret_key_share(index);
     DkgResult::new(secret_key_set.public_keys(), Some(secret_key_share))
 }
 
+/// Generate a BLS SecretKeySet for the given number of participants.
+/// Used for generating first node, or for test.
+pub fn generate_bls_threshold_secret_key(
+    rng: &mut MainRng,
+    participants: usize,
+) -> RealBlsSecretKeySet {
+    let threshold = participants.saturating_sub(1) / 3;
```

Comment:
better have `3` as a const ?

Reply:
Went for comment instead seemed more useful.